plugins/vpc-shared-eni/plugin/commands.go

```diff
@@ -100,10 +118,10 @@ func (plugin *Plugin) Add(args *cniSkel.CmdArgs) error {
 		},
 		IPs: []*cniTypesCurrent.IPConfig{
```

Comment:
We need to populate one IPConfig entry for each IP address here.

Reply:
If the CNI output contains both v4 and v6 addresses kubelet is defaulting to giving the pod an IPv4 address. So i had to remove the v4 address from cni output struct if an ipv6 address is present.

Reply:
@sgundapu: Can you describe that in a comment somewhere near your line 95 code above?  I agree with Onur, most readers are going to see this and think "surely this should be a loop and return an array?"

Reply:
@ofiliz: I don't think we have much choice here atm.  Once https://github.com/containerd/containerd/pull/5964 is in a release that we're using, an alternative option would be to return both v4 and v6 here, but set that new `ip_pref = "ipv6"` (or `= "cni"`) containerd config option.